File: server/config/supabase.js
const { createClient } = require('@supabase/supabase-js');

// Supabase Admin client for server-side operations
// Uses the service_role key for full database access
const supabaseUrl = process.env.SUPABASE_URL || 'https://your-project.supabase.co';
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY || 'your-service-role-key';

let supabase;
try {
  supabase = createClient(supabaseUrl, supabaseServiceKey, {
    auth: {
      autoRefreshToken: false,
      persistSession: false
    }
  });
  console.log('✅ Supabase client initialized');
} catch (error) {
  console.warn('⚠️  Supabase initialization warning:', error.message);
  supabase = null;
}

module.exports = { supabase };

File: backend/config/supabase.py
"""
Supabase client configuration.
Initializes the admin client using the service role key for server-side operations.
"""

import os
from dotenv import load_dotenv
from supabase import create_client, Client

# Ensure .env is loaded before reading env vars
load_dotenv()

supabase_url: str = os.getenv("SUPABASE_URL", "")
supabase_service_key: str = os.getenv("SUPABASE_SERVICE_ROLE_KEY", "")

supabase: Client | None = None

if supabase_url and supabase_service_key and "your-" not in supabase_url:
    try:
        supabase = create_client(supabase_url, supabase_service_key)
        print(f"Supabase connected: {supabase_url}")
    except Exception as e:
        print(f"Supabase initialization warning: {e}")
        supabase = None
else:
    print("Supabase not configured — running in demo mode (in-memory storage)")


def get_supabase() -> Client | None:
    """Return the initialized Supabase client."""
    return supabase
